scanner/indexer/indexer.go

```diff
@@ -97,12 +97,22 @@ func newLibindex(ctx context.Context, indexerCfg config.IndexerConfig, store cci
 		ScanLockRetry:        libindex.DefaultScanLockRetry,
 		LayerScanConcurrency: libindex.DefaultLayerScanConcurrency,
 	}
-
-	opts.ScannerConfig.Repo = map[string]func(interface{}) error{
-		"rhel-repository-scanner": deserializeFunc([]byte(fmt.Sprintf(`{"repo2cpe_mapping_url": "%s"}`, indexerCfg.RepositoryToCPEURL))),
-	}
-	opts.ScannerConfig.Package = map[string]func(interface{}) error{
-		"rhel_containerscanner": deserializeFunc([]byte(fmt.Sprintf(`{"name2repos_mapping_url": "%s"}`, indexerCfg.NameToCPEURL))),
+	err = sanityCheckURL(indexerCfg.RepositoryToCPEURL)
+	if err == nil {
+		opts.ScannerConfig.Repo = map[string]func(interface{}) error{
+			"rhel-repository-scanner": deserializeFunc([]byte(fmt.Sprintf(`{"repo2cpe_mapping_url": "%s"}`, indexerCfg.RepositoryToCPEURL)))}
+		opts.ScannerConfig.Package = map[string]func(interface{}) error{
+			"rhel_containerscanner": deserializeFunc([]byte(fmt.Sprintf(`{"name2repos_mapping_url": "%s"}`, indexerCfg.NameToCPEURL)))}
+	} else { // if central or sensor is not available
+		zlog.Info(ctx).Err(err).Msg("Mapping file endpoints are currently unavailable")
+		opts.ScannerConfig.Repo = map[string]func(interface{}) error{
+			"rhel-repository-scanner": deserializeFunc([]byte(fmt.Sprintf(
+				`{"repo2cpe_mapping_url": "%s","repo2cpe_mapping_file": "%s"}`, indexerCfg.RepositoryToCPEURL, "mappingfiles/repository-to-cpe.json"))),
```

Comment:
Just to confirm: After the initial pull, the next one will reach out to the URL, correct?

Reply:
yep: https://github.com/quay/claircore/blob/main/rhel/repositoryscanner.go#L151

Reply:
Thanks.